### DISCUSSION TITLE
pkg_config: hard-coded path in Libs.private

### DISCUSSION THREAD

**Discussion by dasergatskov** [Homebrew/discussions#5739] [Everyday usage]
### Output of `brew config`

```shell
% brew config
HOMEBREW_VERSION: 4.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 254bf3fe9d8fa2e1b2fb55dbcf535b2d870180c4
Last commit: 6 days ago
Core tap JSON: 16 Nov 22:05 UTC
Core cask tap JSON: 16 Nov 22:05 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 14
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 14-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
% brew doctor
Your system is ready to brew.
```

### Description of issue

In many *.pc files Libs.private has a hardcoded version number in the library path. E.g. 
```
% cat /opt/homebrew/lib/pkgconfig/GraphicsMagick.pc

prefix=/opt/homebrew/Cellar/graphicsmagick/1.3.45_1
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include/GraphicsMagick


Name: GraphicsMagick
Version: 1.3.45
Description: GraphicsMagick image processing library
URL: http://www.graphicsmagick.org/
Libs: -L${libdir} -lGraphicsMagick
Libs.private: -L/opt/homebrew/opt/zstd/lib -L/opt/homebrew/Cellar/little-cms2/2.16/lib -L/opt/homebrew/opt/libpng/lib -L/opt/homebrew/Cellar/jpeg-turbo/3.0.3/lib -L/opt/homebrew/opt/freetype/lib -L/opt/homebrew/Cellar/webp/1.4.0/lib -L/opt/homebrew/Cellar/libtiff/4.6.0/lib -L/opt/homebrew/Cellar/jpeg-xl/0.11.0/lib -L/opt/homebrew/Cellar/libheif/1.18.2/lib -L/opt/homebrew/Cellar/jasper/4.2.4/lib -llcms2 -lfreetype -lbz2 -lz -lltdl -lm -lpthread 
Cflags: -I${includedir} 
```
Since some of those packages have been upgraded the "-L..." path is just invalid and hence the *.pc file is broken.
(E.g. currently libheif is 1.19.3, libtiff is 4.7.0, etc...)

**Comment by ZhongRuoyu** [comment]
This is a somewhat known issue that needs to be fixed manually for the time being. @cho-m has some exciting work https://github.com/Homebrew/brew/pull/18229 to make this no longer necessary, but I'll fix the individual issues seen here for now:

- https://github.com/Homebrew/homebrew-core/pull/197956
- https://github.com/Homebrew/homebrew-core/pull/197957
- https://github.com/Homebrew/homebrew-core/pull/197958
- https://github.com/Homebrew/homebrew-core/pull/197959
- https://github.com/Homebrew/homebrew-core/pull/197960
- https://github.com/Homebrew/homebrew-core/pull/197962
- https://github.com/Homebrew/homebrew-core/pull/197963
- https://github.com/Homebrew/homebrew-core/pull/197964
